lib/Sabberworm/CSS/Parser.php

```diff
@@ -56,7 +59,7 @@ public function __construct($sText, Settings $oParserSettings = null) {
 
 	public function setCharset($sCharset) {
 		$this->sCharset = $sCharset;
-		$this->iLength = $this->strlen($this->sText);
+		$this->iLength = count($this->aText);
```

Comment:
I see a problem here: splitting should only happen after the charset is known (because it depends on the charset used).